Content/posts/auto-generating-auto-hosting-and-auto-updating-docc-documentation.md

```diff
@@ -0,0 +1,115 @@
+
+### Adding the Swift DocC plugin
+
+You may already have taken this first step if you’ve worked with DocC locally. If not, then add the following lines to the end of your `Package.swift`:
+
+```swift
+#if swift(>=5.6)
+  // Add the documentation compiler plugin if possible
+  package.dependencies.append(
+    .package(url: “https://github.com/apple/swift-docc-plugin“, from: “1.0.0“)
+  )
+#endif
+```
+
+Making the plugin dependency conditional means that nothing will break even if the `swift-tools-version` of your manifest is for an earlier version of Swift than 5.6. Naturally, this is unnecessary if your package _only_ supports Swift 5.6 and above.
+
+### Add a Swift Package Index manifest file
+
+Then, create and commit a manifest file named `.spi.yml` file in the root of your repository that looks something like this:
+
+```yaml
+version: 1
+builder:
+  configs:
+    - documentation_targets: [Target]
+```
+
+This file tells our build system to generate documentation for the target `Target`. You would typically set this as your package’s main (or only) target, but it may also be a dedicated target containing documentation Markdown files.
+
+You can also specify multiple targets, and we’ll add a target switcher in the hosted documentation so people can easily find _all_ your documentation!
+
+## Documentation platform
+
+By default, we will generate documentation using macOS. If your package requires the documentation generation to be run for a certain platform such as iOS, you can also specify a platform:
+
+```yaml
+version: 1
+builder:
+  configs:
+    - platform: ios
+      documentation_targets: [Target]
+```
+
+## Auto-updating frequency
+
+To keep the amount of processing that our build servers perform under control, we only build the default branch for each package at most once every 24 hours. So, when you push the configuration file live, the system will generate that set of documentation, but it will then be 24 hours until the generation process runs again. If there have been any commits during that period, we’ll create docs from the latest commit when the period resets.
+
+<picture class="shadow">
+  <source srcset="/images/hosted-docc-documentation~dark.png" media="(prefers-color-scheme: dark)">
+  <img src="hosted-docc-documentation~dark.png" alt="Hosted documentation for the DocC package shown in the context of the Swift Package Index with a header above the documentation.">
+</picture>
+
+## Early adopters!
+
+You may have seen a [call for package authors with DocC compatible documentation](https://twitter.com/SwiftPackages/status/1531299947462676480) earlier this week, and I’m thrilled to say that we have over XXX packages that have already added configuration files and have their documentation hosted by us! Why not check them out? We are currently generating documentation for nineteen packages:
```

Comment:
don't forget to replace the XXX placeholder here!

Reply:
Tell me how you know I worked on this part of the post several days ago without telling me how you know I worked on this part of the post several days ago 😂

That is extremely well spotted, @Sherlouk! Thank you

Reply:
It's why you always want fresh eyes to do the final review! Untainted.